common/djangoapps/course_modes/views.py

```diff
@@ -241,18 +241,36 @@ def post(self, request, course_id):
         allowed_modes = CourseMode.modes_for_course_dict(course_key)
         if requested_mode not in allowed_modes:
             return HttpResponseBadRequest(_("Enrollment mode not supported"))
-
-        if requested_mode == 'audit':
+        if requested_mode in {'audit', 'honor'}:
```

Comment:
Does it make sense to instead reference [CourseMode.AUDIT_MODES](https://github.com/edx/edx-platform/blob/master/common/djangoapps/course_modes/models.py#L137) here?